From 459a5d7d2d8c90168dba26e7a9110d4fb649c217 Mon Sep 17 00:00:00 2001
From: Mickey Sola <msola@sourcefire.com>
Date: Fri, 27 Oct 2017 17:24:33 -0400
Subject: bb11940 - fixing heap overflow in rfc2037. Patch submitted by Suleman
 Ali

Patch-Name: bb11940-fixing-heap-overflow-in-rfc2037.-Patch-submi.patch
---
 libclamav/mbox.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libclamav/mbox.c b/libclamav/mbox.c
index 8e48bb7..13edb78 100644
--- a/libclamav/mbox.c
+++ b/libclamav/mbox.c
@@ -2842,7 +2842,7 @@ rfc2047(const char *in)
 		memcpy(pout, blobGetData(b), len);
 		blobDestroy(b);
 		messageDestroy(m);
-		if(pout[len - 1] == '\n')
+		if(len > 0 && pout[len - 1] == '\n')
 			pout += len - 1;
 		else
 			pout += len;
